dulwich/porcelain.py

```diff
@@ -1753,6 +1755,136 @@ def update_head(repo, target, detached=False, new_branch=None):
             r.refs.set_symbolic_ref(b"HEAD", to_set)
 
 
+def unstage(repo: Repo, paths: list[bytes] = []):
+    '''
+    unstage specific file in the index
+    Args:
+        repo: dulwich Repo object
+        paths: a list of file to unstage
+    '''
+    for path in paths:
+        index = repo.open_index()
+        tree_id = repo[b'HEAD']._tree
+        try:
+            tree_entry = repo[tree_id].lookup_path(lambda x: repo[x], path)
+        except KeyError:
+            # if tree_entry didnt exist, this file was being added, so remove index entry
+            try:
+                del index[path]
+                index.write()
+            except KeyError:
+                print('file not in index.', path.decode())
```

Comment:
this should probably raise an exception?

Reply:
It will raise KeyError in the new commit